src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -33,30 +33,40 @@ namespace mpl = multipass::logging;
 namespace
 {
 constexpr auto category = "sshfs mount";
-template <typename Callable>
-auto run_cmd(mp::SSHSession& session, std::string&& cmd, Callable&& error_handler)
+auto run_cmd(mp::SSHSession& session, std::string&& cmd)
 {
     auto ssh_process = session.exec(cmd);
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
-    return ssh_process.read_std_output();
-}
+        throw std::runtime_error(ssh_process.read_std_error());
 
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
-{
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    return ssh_process.read_std_output();
 }
 
-void check_sshfs_exists(mp::SSHSession& session)
+mp::SshfsMount::SshfsPkgType get_sshfs_package_type(mp::SSHSession& session)
 {
-    auto error_handler = [](mp::SSHProcess& proc) {
+    try
+    {
+        // Prefer to use snap package version first
+        run_cmd(session, "snap list multipass-sshfs");
+        return mp::SshfsMount::SshfsPkgType::snap;
+    }
+    catch (const std::exception& e)
+    {
+        mpl::log(mpl::Level::debug, category, fmt::format("'sshfs' snap package is not installed: {}", e.what()));
+    }
+
+    try
+    {
+        // Fallback to looking for Debian version if snap is not found
+        run_cmd(session, "which sshfs");
+        return mp::SshfsMount::SshfsPkgType::debian;
+    }
```

Comment:
If anything, this should be more generic than "debian", since this may be any other distro.

Reply:
Well, it's just an internal naming implementation, so I don't think the name of the enum value is important here.  That said, I'd rather just get rid of it altogether as mentioned above.

Reply:
It's gone, so no longer an issue.